chain/indexer/src/block.rs

```diff
@@ -151,116 +151,126 @@ where
+                            Ok(current_chain_block_number) => {
+                                chain_head = current_chain_block_number;
+                            }
+                            Err(error) => {
+                                error!("failed to fetch block number from RPC: {error}");
+                                chain_head = chain_head.max(current_block);
+                            }
+                        }
 
-                match rpc.block_number().await {
-                    Ok(current_chain_block_number) => {
-                        chain_head = current_chain_block_number;
-                    }
-                    Err(error) => {
-                        error!("failed to fetch block number from RPC: {error}");
-                        chain_head = chain_head.max(current_block);
-                    }
-                }
+                        if !is_synced.load(std::sync::atomic::Ordering::Relaxed) {
+                            let progress =
+                                (current_block - latest_block_in_db) as f64 / (chain_head - latest_block_in_db) as f64;
+                            info!("Sync progress {:.2}% @ block {}", progress * 100_f64, current_block);
 
-                if tx.is_some() {
-                    let progress =
-                        (current_block - latest_block_in_db) as f64 / (chain_head - latest_block_in_db) as f64;
-                    info!("Sync progress {:.2}% @ block {}", progress * 100_f64, current_block);
+                            #[cfg(all(feature = "prometheus", not(test)))]
+                            METRIC_INDEXER_SYNC_PROGRESS.set(progress);
 
-                    #[cfg(all(feature = "prometheus", not(test)))]
-                    METRIC_INDEXER_SYNC_PROGRESS.set(progress);
+                            if current_block >= chain_head {
+                                info!("Indexer sync successfully completed");
+                                is_synced.store(true, std::sync::atomic::Ordering::Relaxed);
+                                if let Err(e) = tx.try_send(()) {
+                                    error!("failed to notify about achieving index synchronization: {e}")
+                                }
+                            }
+                        }
 
-                    if current_block >= chain_head {
-                        info!("Indexer sync successfully completed");
-                        let _ = tx.take().expect("index sync finalization should be present").send(());
+                        block_with_logs
+                    }
+                })
+                .flat_map(|mut block_with_logs| {
+                    // Assuming sorted and properly organized blocks,
+                    // the following lines are just a sanity safety mechanism
+                    block_with_logs.logs.sort_by(log_comparator);
+                    stream::iter(block_with_logs.logs.into_iter().map(move |v| v))
```

Comment:
The identity map on the iterator looks redundant.

Reply:
Indeed it is... that's why the clippy fails.